packages/instrumentor/instrument.ts

```diff
@@ -110,22 +129,69 @@ export function shouldInstrumentFn(
 	};
 }
 
+function shouldCollectCoverageFn(
+	coverage: boolean,
+	customHooks: string[],
+	shouldInstrument: FilePredicate = () => true
+): FilePredicate {
+	const shouldCoverCustomHooks = shouldInstrumentFn(customHooks, ["nothing"]);
+	return (filepath: string) => {
+		return (
+			coverage &&
+			(shouldCoverCustomHooks(filepath) || shouldInstrument(filepath))
+		);
+	};
+}
+
 function instrument(
 	code: string,
 	filename: string,
 	shouldInstrument: FilePredicate,
-	shouldHook: FilePredicate
+	shouldHook: FilePredicate,
+	shouldCollectCoverage: FilePredicate,
+	dryRun: boolean
 ) {
 	const transformations: PluginItem[] = [];
-	if (shouldInstrument(filename)) {
-		transformations.push(codeCoverage, compareHooks);
+	if (!dryRun) {
+		if (shouldInstrument(filename)) {
+			transformations.push(codeCoverage, compareHooks);
+		}
 	}
 	if (shouldHook(filename)) {
 		transformations.push(functionHooks(filename));
 	}
-	return transform(filename, code, transformations)?.code || code;
+	if (shouldCollectCoverage(filename)) {
+		transformations.push(istanbulCodeCoverage(filename));
+	}
+
+	// hook stderr before calling babel to avoid printing warnings to the console
+	const { fnReturnValue } = callWithHookedStderr(() => {
```

Comment:
Why should we silently hide babel errors? Also, we don't seem to do anything with the returned value here

Reply:
It prints many babel errors if the includes and excludes to be instrumented are not set properly. E.g., babel trying to instrument babel.
I moved it into my "future" vault :)
If users complain, we can add it back in in verbose mode.